hotel/models.py

```diff
@@ -0,0 +1,137 @@
+from hotel import *
+
+
+def _night(name):
+    day = getattr(c, name.upper())
+
+    def lookup(self):
+        return day if day in self.nights_ints else ''
+    lookup.__name__ = name
+    lookup = property(lookup)
+
+    def setter(self, val):
+        if val:
+            self.nights = '{},{}'.format(self.nights, day).strip(',')
+        else:
+            self.nights = ','.join([str(night) for night in self.nights_ints if night != day])
+    setter.__name__ = name
+
+    return lookup.setter(setter)
+
+
+class NightsMixin(object):
+    @property
+    def nights_labels(self):
+        ordered = sorted(self.nights_ints, key=c.NIGHT_DISPLAY_ORDER.index)
+        return [c.NIGHTS[val] for val in ordered]
+
+    @property
+    def nights_display(self):
+        return ' / '.join(self.nights_labels)
+
+    @property
+    def setup_teardown(self):
+        return any(night for night in self.nights_ints if night not in c.CORE_NIGHTS)
+
+    locals().update({mutate(name): _night(mutate(name)) for name in c.NIGHT_NAMES for mutate in [str.upper, str.lower]})
+
+
+@Session.model_mixin
+class Attendee:
```

Comment:
Previously, we defined hotel eligibility as "do you have a Staff badge?"  Now there's an actual boolean field for it, which is set to true automatically if someone has a Staff badge.  So the behavior is basically unchanged, but now it can be overridden if we want to e.g. stick a Guest into a Staff room or something.